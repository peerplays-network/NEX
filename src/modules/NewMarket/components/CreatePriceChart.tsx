import React from "react";

import { useCreatePriceChart } from "./hooks/useCreatePriceChart";

const CreatePriceChart = (): JSX.Element => {
  const { chartContainerRef } = useCreatePriceChart();

  return <div ref={chartContainerRef} />;
};

export default CreatePriceChart;
