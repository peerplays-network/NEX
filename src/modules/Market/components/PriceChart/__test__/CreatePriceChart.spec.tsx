import { cleanup, render } from "@testing-library/react";
import React from "react";

import CreatePriceChart from "../CreatePriceChart";

jest.mock("../hooks/useCreatePriceChart.ts", () => ({
  useCreatePriceChart: jest.fn().mockReturnValue({
    chartContainerRef: React.createRef(),
  }),
}));

describe("CreatePriceChart component", () => {
  afterEach(() => {
    cleanup();
    jest.resetAllMocks();
  });

  it("should render the component correctly", () => {
    const { container, asFragment } = render(<CreatePriceChart />);

    expect(asFragment()).toMatchSnapshot();
    expect(container.firstChild).toBeDefined();
  });
});
