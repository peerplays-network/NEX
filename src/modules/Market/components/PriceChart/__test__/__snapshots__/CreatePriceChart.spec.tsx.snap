// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePriceChart component should render the component correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe kugChA"
  />
</DocumentFragment>
`;
